R/source.R

```diff
@@ -253,21 +253,33 @@ invoke_cargo <- function(toolchain, specific_target, dir, profile,
       )
     }
 
-    # rtools_path() returns path to the RTOOLS40_HOME\usr\bin,
-    # but we need RTOOLS40_HOME\mingw{arch}\bin, hence the "../.."
-    rtools_home <- normalizePath(
-      # `pkgbuild` may return two paths for R < 4.2 with Rtools40v2
-      file.path(pkgbuild::rtools_path()[1], "..", ".."),
-      winslash = "/",
-      mustWork = TRUE
-    )
-
     if (identical(R.version$crt, "ucrt")) {
+      # TODO: update this when R 5.0 is released.
+      if (!identical(R.version$major, "4")) {
+        ui_throw("rextendr currently supports R 4.x")
+      }
+
+      if (package_version(R.version$minor) >= "3.0") {
+        rtools_version <- "43"  # nolint: object_usage_linter
+      } else {
+        rtools_version <- "42"  # nolint: object_usage_linter
+      }
+
+      rtools_home <- Sys.getenv(glue("RTOOLS{rtools_version}_HOME"),
```

Comment:
Should we also use `normalizePath` here and require it to exist (i.e. `mustWork = TRUE`)?